experiments/AtmosLES/ekman_layer_model.jl

```diff
@@ -0,0 +1,298 @@
+    add_perturbations!(state, localgeo)
+    init_state_prognostic!(bl.turbconv, bl, state, aux, localgeo, t)
+end
+
+function ekman_layer_model(
+    ::Type{FT},
+    config_type,
+    zmax,
+    surface_flux;
+    turbulence = ConstantKinematicViscosity(FT(0.1)),
+    turbconv = NoTurbConv(),
+    compressibility = Compressible(),
+    ref_state = HydrostaticState(DryAdiabaticProfile{FT}(param_set),),
+) where {FT}
+
+    ics = init_problem!     # Initial conditions
+
+    C_drag_::FT = C_drag(param_set) # FT(0.001)    # Momentum exchange coefficient
+    u_star = FT(0.30)
+    z_0 = FT(0.1)          # Roughness height
+
+    z_sponge = FT(300)     # Start of sponge layer
+    α_max = FT(0.75)       # Strength of sponge layer (timescale)
+    γ = 2                  # Strength of sponge layer (exponent)
+
+    u_geostrophic = FT(1)        # Eastward relaxation speed
+    u_slope = FT(0)              # Slope of altitude-dependent relaxation speed
+    v_geostrophic = FT(0)        # Northward relaxation speed
+    f_coriolis = FT(1.39e-4) # Coriolis parameter at 73N
+
+    q_sfc = FT(0)
+    θ_sfc = T_surf_ref(param_set)
+    g = compressibility isa Compressible ? (Gravity(),) : ()
+
+    # Assemble source components
+    source_default = (
+        g...,
+        EkmanLayerSponge{FT}(
+            zmax,
+            z_sponge,
+            α_max,
+            γ,
+            u_geostrophic,
+            u_slope,
+            v_geostrophic,
+        ),
+        EkmanLayerGeostrophic(
+            f_coriolis,
+            u_geostrophic,
+            u_slope,
+            v_geostrophic,
+        ),
+        turbconv_sources(turbconv)...,
+    )
+    source = source_default
+
+    # Set up problem initial and boundary conditions
+    if surface_flux == "prescribed"
+        energy_bc = PrescribedEnergyFlux((state, aux, t) -> FT(0))
+        moisture_bc = PrescribedMoistureFlux((state, aux, t) -> FT(0))
+    elseif surface_flux == "bulk"
+        energy_bc = BulkFormulaEnergy(
+            (bl, state, aux, t, normPu_int) -> C_drag_,
+            (bl, state, aux, t) -> (θ_sfc, q_sfc),
+        )
+        moisture_bc = BulkFormulaMoisture(
+            (state, aux, t, normPu_int) -> C_drag_,
+            (state, aux, t) -> q_sfc,
+        )
+    elseif surface_flux == "custom_sbl"
+        energy_bc = PrescribedTemperature((state, aux, t) -> θ_sfc)
+        moisture_bc = BulkFormulaMoisture(
+            (state, aux, t, normPu_int) -> C_drag_,
+            (state, aux, t) -> q_sfc,
+        )
+    elseif surface_flux == "Nishizawa2018"
+        energy_bc = NishizawaEnergyFlux(
+            (bl, state, aux, t, normPu_int) -> z_0,
+            (bl, state, aux, t) -> (θ_sfc, q_sfc),
+        )
+        moisture_bc = PrescribedMoistureFlux((state, aux, t) -> FT(0))
+    else
+        @warn @sprintf(
+            """
+%s: unrecognized surface flux; using 'prescribed'""",
+            surface_flux,
+        )
+    end
+
+    moisture_bcs = ()
```

Comment:
why is this here if 'moisture_bc' is defined above ?

Reply:
good catch